q2_fondue/entrezpy_clients/_efetch.py

```diff
@@ -541,4 +530,8 @@ def analyze_result(self, response, request):
     def parse(self, raw_response, request):
         response = self.convert_response(
             raw_response.read().decode('utf-8'), request)
-        self.analyze_result(response, request)
+        if self.isErrorResponse(response, request):
+            self.hasErrorResponse = True
+            self.analyze_error(response, request)
+        else:
+            self.analyze_result(response, request)
```

Comment:
Since the default `parse` method calls a `sys.exit` if `self.result is None` and our implementation only initializes `self.result` if the response is not an error in `self.analyze_result`, I suggest we keep the custom `parse` method for now. I am happy to adjust the comment above the custom `parse` method accordingly.

Reply:
As for `xml` vs. `text`, I also think that everything works the same :)

Reply:
as for `xml` vs `text`: yes, I also think they return the same result.